src/DeFuncto.Tests/Core/Extensions/Tasks/RunAsync.cs

```diff
@@ -0,0 +1,21 @@
+﻿using DeFuncto.Assertions;
+using DeFuncto.Extensions;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace DeFuncto.Tests.Core.Extensions.Tasks
+{
+    public class RunAsync
+    {
+
+        [Fact(DisplayName = "Runs an asynchronous action")]
+        public async Task Runs()
+        {
+            var witness = new Witness();
+            await witness.RunAsync(async w => w.Touch());
```

Comment:
Holy, this checks are quite cool.
Reintroduce the pragmas.
Perhaps only inside the lambdas. try and see if it can be done.